### ISSUE TITLE
Single Corrupt Record Breaks Scan

### ISSUE THREAD

**Issue by SteggyLeggy** [pynamodb/PynamoDB#1067]
We currently have an issue that our current API doesn't support the same number of fields that we support within our DynamoDB records. So we have a lot of employees manually editing/creating records within the DynamoDB Tables by hand.

Occasionally they forget to add required fields, or enter the wrong data for a field. These employees are in a separate department so sometimes they're not aware of new fields added recently.

Some of our tables only contain a few hundred records, and our production systems routinely scan the full table, using PynamoDB. This means that a single corrupt record (i.e. missing a field) raises an exception and we cannot do any work required on the remaining valid records.

Is there a way in PynamoDB to defer deserialization until later on, i.e. allow the scan to happen, but deserialize each record individually so that we can catch the exception and ignore the problem records?

Would really appreciate some help on this, we have some scope to help improve PynamoDB in this scenario if there currently isn't a way to support/skip over problem records currently.

**Comment by SteggyLeggy** [comment]
Thanks for your suggestion @mateuszciosek but unfortunately it isn't just missing fields that is the problem.

We have more complex Map fields that aren't formatted correctly when edited manually in the aws console.

Trying to make the pynamoDB model less strict isn't really what we're after. More just letting us skip over the dodgy records.

Thanks again for your help though.

**Comment by ikonst** [comment]
That's a great question. Perhaps we can add a `try: bool = False` parameter that'll cause the iteration to be over `Union[Model, DeserializationFailure]` where `DeserializationFailure` could include:
- `raw_data: Dict[str, Any]`
- `deserialization_exception: Exception`

Another thing I was thinking about, is to allow models to be defined as lazily deserializable, mostly for deserialization performance of non-corrupt models, but it might help here too.

**Comment by ikonst** [comment]
Yeah, that might work. Make sure you keep some way in your model to indicate that it wasn't initialized.
